rllab/envs/mujoco/gather/embedded_viewer.py

```diff
@@ -35,52 +39,67 @@ def __init__(self):
     def set_model(self, model):
         self.model = model
         if model:
-            self.data = model.data
+            # self.data = model.data
+            self.sim = MjSim(self.model)
+            self.data = self.sim.data
+            self.con = MjRenderContext(self.sim)
+            self.scn = self.con.scn
+            self.vopt = self.con.vopt
         else:
             self.data = None
         if self.running:
             if model:
-                mjlib.mjr_makeContext(model.ptr, byref(self.con), 150)
+                # functions.mjr_makeContext(model.ptr, byref(self.con), 150)
+                functions.mjr_makeContext(self.model, self.con, 150)
             else:
-                mjlib.mjr_makeContext(None, byref(self.con), 150)
+                # functions.mjr_makeContext(None, byref(self.con), 150)
+                functions.mjr_makeContext(None, self.con, 150)
             self.render()
         if model:
             self.autoscale()
 
     def autoscale(self):
-        self.cam.lookat[0] = self.model.stat.center[0]
-        self.cam.lookat[1] = self.model.stat.center[1]
-        self.cam.lookat[2] = self.model.stat.center[2]
-        self.cam.distance = 1.0 * self.model.stat.extent
+        self.cam.lookat[0] = self.sim.stat.center[0]
+        self.cam.lookat[1] = self.sim.stat.center[1]
+        self.cam.lookat[2] = self.sim.stat.center[2]
+        self.cam.distance = 1.0 * self.sim.stat.extent
         self.cam.camid = -1
         self.cam.trackbodyid = -1
         if self.window:
             width, height = glfw.get_framebuffer_size(self.window)
-            mjlib.mjv_updateCameraPose(byref(self.cam), width * 1.0 / height)
+            # Unsure if API translates
```

Comment:
Ah. If you are unsure whether these APIs translate properly you should test that :).

If the API does not translate, and it's a big problem to fix, then get it to work well enough with the current API and leave a TODO/GitHub issue to fix later.

Reply:
The only reason I bring it up is because I could not find any examples that utilize `GatherViewer`, which is the only place that utilizes this `EmbeddedViewer` file.